src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -72,8 +72,10 @@ public static function createResourceForClass(\ReflectionClass $reflectionClass)
             $metadata['__construct'] = self::getResourceMetadataForMethod($constructor);
         }
 
-        foreach (self::getSetters($reflectionClass) as $reflectionMethod) {
-            $metadata[$reflectionMethod->name] = self::getResourceMetadataForMethod($reflectionMethod);
+        foreach ($reflectionClass->getMethods(\ReflectionMethod::IS_PUBLIC) as $reflectionMethod) {
```

Comment:
We can indeed allow masks in the configuration, but it's not related to this code block (it is not aware of the configuration so it needs to watch all methods anyway).